src/local_view/elements/computation_node.ts

```diff
@@ -0,0 +1,194 @@
+                index: idx,
+            });
+            const prev = idxMap.get(val.dataContainer);
+            if (prev !== undefined)
+                prev.push([val.accessMode, idx]);
+            else
+                idxMap.set(val.dataContainer, [[val.accessMode, idx]]);
+        }
+
+        return [idxMap, resolvedAccessOrder];
+    }
+
+    private findRelatedFromScope(
+        source: DataContainer, scope: Map<string, number>,
+        idxMap: AccessMap<(number | undefined)[]>
+    ): void {
+        for (const access of this.accessOrder) {
+            if (!RELATE_INCLUSIVE && access.dataContainer === source)
+                continue;
+            if (access.index) {
+                const idx: (number | undefined)[] = [];
+                for (const e of access.index) {
+                    let res = undefined;
+                    try {
+                        res = math.evaluate(e.replaceAll('_', ''), scope);
+                        if (typeof res !== 'number')
+                            res = undefined;
+                    } catch (_ignored) {
+                        res = undefined;
+                    }
+                    idx.push(res);
+                }
+                const prev = idxMap.get(access.dataContainer);
+                if (prev !== undefined)
+                    prev.push([access.accessMode, idx]);
+                else
+                    idxMap.set(
+                        access.dataContainer,
+                        [[access.accessMode, idx]]
+                    );
+            }
+        }
+    }
+
+    /**
+     * For a given data container and numeric index, get all related accesses.
+     * @param source    Source data container
+     * @param index     Numeric index in source data container
+     * @param origin    The node asking for related accesses
+     * @returns         Access map of related accesses
+     */
+    public getRelatedAccesses(
+        source: DataContainer, index: number[], origin?: Node
+    ): AccessMap<(number | undefined)[]> {
+        const idxMap = new AccessMap<(number | undefined)[]>();
+
+        // Find out what symbolic accesses the numeric indices to the source
+        // relate to.
+        const sourceAccesses: string[][] = [];
+        this.accessOrder.forEach(val => {
+            if (val.dataContainer === source)
+                sourceAccesses.push(val.index);
+        });
+
+        // Construct a scope which reflects the symbol values that result in
+        // this data access. We can use this to deduce further data accesses
+        // based on symbolic indices.
+        if (sourceAccesses.length > 0) {
+            const scope = new Map<string, number>();
+            sourceAccesses[0].forEach((idx: string, i: number) => {
+                if (i < index.length) {
+                    const rightHand = index[i];
+                    const leftHand = idx;
+                    // TODO: Having to replace underscores here is ugly. We
```

Comment:
CoffeQuate didn't allow underscores at the start of variable names (testing implied that it didn't allow them at all, but I now found out it is only not allowed at the beginning). I have now provided a patched version of CoffeeQuate to avoid this issue, so this code could be removed.